### PR TITLE
raft: strongly type IDs

### PR DESCRIPTION
**Pull request by arulajmani**
See individual commits for details.

### CODE REVIEW

**Comment by arulajmani** [comment]
Third commit is unrelated and I don't care too much about. I'd like to get the first one in soon though -- it's mostly mechanical.

**Comment by pav-kv** [comment]
@arulajmani Do you plan to eventually make `type ReplicaID = raftpb.PeerID`? Many casts would be avoided this way, but as we discussed `ReplicID` is int32 while raft ID is uint64, so this would need some alignment.

File: pkg/raft/status.go [hunk 1]
```diff
@@ -78,6 +99,22 @@ func getStatus(r *raft) Status {
 	return s
 }
 
+// getSparseStatus gets a sparse[*] copy of the current raft status.
+//
+// [*] See struct definition for what this entails.
+func getSparseStatus(r *raft) SparseStatus {
```
Comment (pav-kv) [inline pkg/raft/status.go:105]: nit: make this and `withProgress` a method of `*raft`?

**Review by pav-kv — APPROVED** [review]

**Review by arulajmani — COMMENTED** [review]
> Do you plan to eventually make `type ReplicaID = raftpb.PeerID`? Many casts would be avoided this way, but as we discussed `ReplicID` is int32 while raft ID is uint64, so this would need some alignment.

Another option would be to change either `raftpb.PeerID` to be an int32. If I'm reading https://protobuf.dev/programming-guides/proto2/#updating correctly, this should be allowed. Conversely, we could also change `ReplicaID` to be a uint64. Let's start a thread on this in kv global. 

 TFTR!

bors r=pav-kv

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/126343#-:-O0UNRjf74Jzk5pEoMqP:b-sw5qqp)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @kyle-a-wong, @nvanbenschoten, and @pav-kv)

___
*[`pkg/raft/status.go` line 105 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/126343#gh-1658690051:-O0UO2u29KlL_G0OjAKJ:b-sm4lpk) ([raw file](https://github.com/cockroachdb/cockroach/blob/fb05fcf8ec2a92e3422d9ddd30fd333bea171fd1/pkg/raft/status.go#L105)):*
<details><summary><i>Previously, pav-kv (Pavel Kalinnikov) wrote…</i></summary><blockquote>

nit: make this and `withProgress` a method of `*raft`?
</blockquote></details>

I'm following the pattern for `getStatus` and `getBasicStatus` here, so I'll keep it this way for now.

**Comment by pav-kv** [comment]
@arulajmani There are some CI failures to fix before the merge. I think something might be up with the redacted formatting for the new type.

**Comment by arulajmani** [comment]
Ah rip. Thanks for calling out. 

bors r-

**Comment by arulajmani** [comment]
CI is green now. I'll merge. 

bors r=pav-kv

**Pull request merged**